packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue

```diff
@@ -157,44 +157,31 @@ function updateModelConfigValue(variableName: string) {
 }
 
 function configureMatrix() {
-	// Get stratified ids from the chosen base id
-	const rowAndCol = props.modelConfiguration.configuration?.semantics?.span?.[0]?.map
-		.filter((id: string) => id[1] === props.id)
-		.map((d) => d[0]);
-
-	// Assign dimensions relevant to the ids
-	const rowAndColDimensions = props.modelConfiguration.configuration.semantics?.typing?.map.filter(
-		(id) => rowAndCol.includes(id[0])
-	);
-
-	if (rowAndColDimensions) {
-		// Assuming column id is the first one
-		const colId = rowAndColDimensions[0][0];
-		let rowId: string | null = null;
-
-		for (let i = 0; i < rowAndColDimensions?.length; i++) {
-			// If other id is recognized it is the row id
-			if (!rowId && rowAndColDimensions[i][0] !== colId) {
-				rowId = rowAndColDimensions[i][0];
-			}
-
-			// Push dimension
-			if (colId === rowAndColDimensions[i][0]) {
-				colDimensions.push(rowAndColDimensions[i][1]);
-			} else if (rowId === rowAndColDimensions[i][0]) {
-				rowDimensions.push(rowAndColDimensions[i][1]);
-			}
-		}
-	}
+	console.log(props.modelConfiguration.configuration);
 
 	// Get only the states/transitions that are mapped to the base model
 	const matrixData =
 		props.nodeType === NodeType.State
-			? extractStateMatrixData(props.modelConfiguration.configuration, rowAndCol, [
-					...colDimensions,
-					...rowDimensions
-			  ])
-			: extractTransitionMatrixData(props.modelConfiguration.configuration, rowAndCol);
+			? getCatlabStatesMatrixData(props.modelConfiguration.configuration).filter(
+					(d) => d['@base'] === props.id
+			  )
+			: getCatlabTransitionsMatrixData(props.modelConfiguration.configuration).filter(
+					(d) => d['@base'] === props.id
+			  );
+
+	console.log(matrixData);
+
+	if (isEmpty(matrixData)) return;
+
+	// Grab dimension names from the first matrix row
+	const dimensions = [cloneDeep(matrixData)[0]].map((d) => {
+		delete d.id;
+		delete d['@base'];
+		return Object.keys(d);
+	})[0];
```

Comment:
By the way, I am not crazy about using the term `@base` ... only using it because the keys are generated and I wanted to avoid collisions if possible. If you have better suggestions I am all ears.